### DISCUSSION TITLE
Several questions about e2e test.

### DISCUSSION THREAD

**Discussion by littletiger123** [kubernetes-sigs/kubebuilder#2259] [Support] [answered]
I am fresh.
I want to write an e2e test on a CRD in minikube rather than envtest which is mentioned in the [page](https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/cronjob-tutorial/testdata/project/controllers/cronjob_controller_test.go). Shall I need to install the project? And is there several demo or guides I can follow up on?

**Comment by camilamacedo86** [comment, marked as answer]
Hi @littletiger123, 

Minikuke is just a vendor for you to have the Kubernetes cluster installed. 

ENV TEST: If a lib to help you out create tests which will be integrated because the requests are actually done against the cluster. However, its purpose is more like "unit" test of the controller's implementation. 

Therefore, I'd recommend checking out the files scaffolded by the new plugin [deploy.image/v1-alpha](https://master.book.kubebuilder.io/plugins/deploy-image-plugin-v1-alpha.html), see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v3-with-deploy-image/controllers/busybox_controller_test.go#L36-L125 as an example.

Also, if you are looking for e2e tests for your whole project you might want to look at the example implemented on the SDK repo, see: https://github.com/operator-framework/operator-sdk/tree/master/testdata/go/v3/memcached-operator/test/e2e. 

Marking it as answered but feel free to create new discussion topics or issues if you need

**Comment by camilamacedo86** [comment]
Closing since it is answered.
